cmd/data-scanner.go

```diff
@@ -59,6 +60,9 @@ const (
 	healDeleteDangling    = true
 	healFolderIncludeProb = 32  // Include a clean folder one in n cycles.
 	healObjectSelectProb  = 512 // Overall probability of a file being scanned; one in n.
+
+	dataScannerExcessiveVersionsThreshold = 1000  // Issue a warning when a single object has more versions than this
```

Comment:
Shouldn't this be a problem even at 100 ? I can't for see anyone needing more than 100 versions

Reply:
100 will not significantly impact it:

https://lh3.googleusercontent.com/KDXCrOlM9NXU2LmxqeH77aC5MALKm1zcwanJJZmdFeR7O4odOpKl2t3QNJ7gesk4E0tQFdyvUa-1Moc1wA5KdNxtN0e1Bq1yo0Iszmq-FVNdkfbSjCjRD5VKGeNsORkf_q0Hsu9w

https://lh4.googleusercontent.com/u90X2L64-haEgfaTjrsNQlguTHRhxXj3qC5Pp0UyW8Mxkp2hEoVnVNQo9M09xNxWLUpfdUYtqQTURRTGU_pZJdorXYgiaBoi9aUS2Ke6KkRkLHSiaAAvIPQqWvPvUTxyv1ldB85P

Reply:
Might as well make it less likely to be noisy when it doesn't really affect us much.

Reply:
These can't be tested in SILO right?

A prefix with many objects each object having 100versions each the cost will eventually add up.

There is no requirement for any application to have anywhere close to 100 versions. Atleast I can't think of any.

Reply:
@harshavardhana It is separate concerns. Many files/folder doesn't affect individual version access. 

Each scales separately, so it isn't a concern if they are combined.

A prefix with 1M objects will be just as slow whether there are 1 or 1000 versions of each. So it makes sense to test them individually and report if either is "unreasonable". Keeping this as non-noisy as possible gives it more value.

We shouldn't dictate usage, except when it affects performance.

Reply:
Okay I think we can live with 1000 for now.